cirq/work/sampler.py

```diff
@@ -14,14 +14,27 @@
 """Abstract base class for things sampling quantum circuits."""
 
 import abc
-from typing import (List, Union)
+from typing import (List, Union, Awaitable)
 
 from cirq import circuits, schedules, study
 
 
 class Sampler(metaclass=abc.ABCMeta):
     """Something capable of sampling quantum circuits. Simulator or hardware."""
 
+    @abc.abstractmethod
+    def async_sample(self, program: Union[circuits.Circuit, schedules.Schedule],
```

Comment:
This is an async version of the `run` method below so I'd suggest calling it `async_run` or `run_async` (I personally prefer the latter with "async" as a suffix so that these two will appear together in autocompletion lists, etc.)

Reply:
Done.